site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -50,14 +50,112 @@ public function editGradeableRequest($gradeable_id, $nav_tab = 0) {
         'electronic_bulk' => 'TA/Instructor will (bulk) upload scanned .pdf for online manual grading'
     ];
 
+    #[Route("/api/{_semester}/{_course}/{gradeable_id}/download", methods: ["GET"])]
+    public function apiDownloadJson(string $gradeable_id): JsonResponse {
+        $gradeable = $this->core->getQueries()->getGradeableConfig($gradeable_id);
+        return JsonResponse::getSuccessResponse($this->getGradeableJson($gradeable));
+    }
+
+    #[Route("/courses/{_semester}/{_course}/{gradeable_id}/download", methods: ["GET"])]
+    public function webDownloadJson(string $gradeable_id): void {
+        $gradeable = $this->core->getQueries()->getGradeableConfig($gradeable_id);
+        $returned_json = $this->getGradeableJson($gradeable);
+        $json_response = JsonResponse::getSuccessResponse($returned_json);
+        // Make the JSON only the data, not the data and the success status.
+        $json_response->json = $json_response->json['data'];
+        $json_response->render($this->core);
+    }
+
+    /**
+     * @return array<mixed>
+     */
+    public function getGradeableJson(Gradeable $gradeable): array {
```

Comment:
This would be a great place to specify the [array shape](https://phpstan.org/writing-php-code/phpdoc-types#array-shapes), so that it is clear exactly what fields are being returned, and what their expected types are.